components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java

```diff
@@ -90,15 +91,15 @@ public static AuthenticatedUser createLocalAuthenticatedUserFromSubjectIdentifie
         AuthenticatedUser authenticatedUser = new AuthenticatedUser();
 
         if (authenticatedSubjectIdentifier.indexOf(CarbonConstants.DOMAIN_SEPARATOR) > 0) {
-            if (UserCoreUtil.getDomainFromThreadLocal() != null && !UserCoreUtil.getDomainFromThreadLocal().isEmpty()) {
-                String[] subjectIdentifierSplits =
-                        authenticatedSubjectIdentifier.split(CarbonConstants.DOMAIN_SEPARATOR, 2);
-                authenticatedUser.setUserStoreDomain(subjectIdentifierSplits[0]);
-                authenticatedUser.setUserName(MultitenantUtils.getTenantAwareUsername(subjectIdentifierSplits[1]));
-            } else {
-                authenticatedUser.setUserName(MultitenantUtils.getTenantAwareUsername(authenticatedSubjectIdentifier));
-            }
+            String[] subjectIdentifierSplits =
+                    authenticatedSubjectIdentifier.split(CarbonConstants.DOMAIN_SEPARATOR, 2);
+            authenticatedUser.setUserStoreDomain(subjectIdentifierSplits[0]);
+            authenticatedUser.setUserName(MultitenantUtils.getTenantAwareUsername(subjectIdentifierSplits[1]));
+        } else if (UserCoreUtil.getDomainFromThreadLocal() != null && !UserCoreUtil.getDomainFromThreadLocal().isEmpty()) {
+            authenticatedUser.setUserStoreDomain(UserCoreUtil.getDomainFromThreadLocal());
+            authenticatedUser.setUserName(MultitenantUtils.getTenantAwareUsername(authenticatedSubjectIdentifier));
         } else {
+            authenticatedUser.setUserStoreDomain(UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME);
```

Comment:
Don't set the default primary domain name as this domain name can be configured as well. There is a Util method at IdentityUtil class to get the primary domain name. Use it.